test/forms/UkimsNumberFormProviderSpec.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms
+
+import forms.behaviours.StringFieldBehaviours
+import org.scalacheck.Gen
+import play.api.data.{Form, FormError}
+
+class UkimsNumberFormProviderSpec extends StringFieldBehaviours {
```

Comment:
Can we test the case where we pass in something that isn't valid data. like AB2343534 or whatever

Reply:
New unit test added